src/os_crypto/md5/md5_op.c

```diff
@@ -14,62 +14,70 @@
 
 #include <stdio.h>
 #include <string.h>
+#include <openssl/evp.h>
 
 #include "md5_op.h"
-#include <openssl/md5.h>
 #include "headers/defs.h"
 #include "headers/file_op.h"
 
 int OS_MD5_File(const char *fname, os_md5 output, int mode)
 {
     FILE *fp;
-    MD5_CTX ctx;
-    unsigned char buf[1024 + 1];
-    unsigned char digest[16];
+    EVP_MD_CTX *mdctx;
+    unsigned char buf[1024];
+    unsigned char digest[EVP_MAX_MD_SIZE];
     size_t n;
 
     memset(output, 0, sizeof(os_md5));
-    buf[1024] = '\0';
 
-    fp = wfopen(fname, mode == OS_BINARY ? "rb" : "r");
+    fp = fopen(fname, mode == OS_BINARY ? "rb" : "r");
```

Comment:
Why did you replace `wfopen`?

Reply:
I took the `fopen` approach from OpenSSL documentation's examples, but we may have extra requirements. I'll revert it back to use `wfopen`.